test/boost/s3_test.cc

```diff
@@ -626,6 +626,44 @@ SEASTAR_THREAD_TEST_CASE(test_object_reupload) {
     }
 }
 
+void test_download_data_source(const client_maker_function& client_maker, unsigned chunks) {
+    const sstring name(fmt::format("/{}/testdatasourceobject-{}", tests::getenv_safe("S3_BUCKET_FOR_TEST"), ::getpid()));
+
+    testlog.info("Make client\n");
+    semaphore mem(16<<20);
+    auto cln = client_maker(mem);
+    auto close_client = deferred_close(*cln);
+
+    static constexpr unsigned chunk_size = 1000;
+    testlog.info("Upload object");
+    auto rnd = tests::random::get_bytes(chunk_size);
+    {
+        auto out = output_stream<char>(cln->make_upload_sink(name));
+        auto close = seastar::deferred_close(out);
+
+        for (unsigned ch = 0; ch < chunks; ch++) {
+            out.write(reinterpret_cast<char*>(rnd.begin()), rnd.size()).get();
+        }
+        out.flush().get();
+    }
+
+    testlog.info("Download object");
+    auto in = input_stream<char>(cln->make_download_source(name, {}, 12345));
+    auto close = seastar::deferred_close(in);
+    for (unsigned ch = 0; ch < chunks; ch++) {
+        auto buf = in.read_exactly(chunk_size).get();
+        BOOST_REQUIRE_EQUAL(memcmp(buf.begin(), rnd.begin(), 1000), 0);
+    }
+}
+
+SEASTAR_THREAD_TEST_CASE(test_download_data_source_minio) {
+    test_download_data_source(make_minio_client, 128 * 1024);
```

Comment:
128K?
Which means the test  will download 128M? (because internally it is 1000 chunk_size is 1000)

For me this is clearer:
```cpp
    static constexpr unsigned chunk_size = 1000;
    static constexpr unsigned file_size = 128000000;
    test_download_data_source(make_minio_client, chunk_size, file_size);
```
and in the function itself
```cpp
auto chunks = file_size / chunk_size
```